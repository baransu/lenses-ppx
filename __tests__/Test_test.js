// Generated by BUCKLESCRIPT VERSION 5.0.6, PLEASE EDIT WITH CARE
'use strict';

var Jest = require("@glennsl/bs-jest/src/jest.js");

function get(state, field) {
  if (field) {
    return state[/* age */1];
  } else {
    return state[/* email */0];
  }
}

function set(state, field, value) {
  if (field) {
    return /* record */[
            /* email */state[/* email */0],
            /* age */value
          ];
  } else {
    return /* record */[
            /* email */value,
            /* age */state[/* age */1]
          ];
  }
}

var State = /* module */[
  /* get */get,
  /* set */set
];

var state = /* record */[
  /* email */"test@test.com",
  /* age */0
];

Jest.describe("lenses-ppx", (function (param) {
        Jest.test("should match snapshot", (function (param) {
                return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](state));
              }));
        Jest.test("should get email", (function (param) {
                return Jest.Expect[/* toBe */2]("test@test.com", Jest.Expect[/* expect */0]("test@test.com"));
              }));
        Jest.test("should get age", (function (param) {
                return Jest.Expect[/* toBe */2](0, Jest.Expect[/* expect */0](0));
              }));
        Jest.test("should set email", (function (param) {
                var match = set(state, /* Email */0, "other@test.com");
                return Jest.Expect[/* toBe */2]("other@test.com", Jest.Expect[/* expect */0](match[/* email */0]));
              }));
        return Jest.test("should set age", (function (param) {
                      var age = set(state, /* Age */1, 10)[/* age */1];
                      return Jest.Expect[/* toBe */2](10, Jest.Expect[/* expect */0](age));
                    }));
      }));

exports.State = State;
exports.state = state;
/*  Not a pure module */
