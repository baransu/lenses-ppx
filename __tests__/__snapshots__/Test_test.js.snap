// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lenses-ppx should match snapshot 1`] = `
Array [
  "test@test.com",
  0,
]
`;
